### PR TITLE
[flang][driver] support -dumpversion and -dumpmachine

### PR DESCRIPTION
**Pull request by yuanfang-chen**
Match GCC driver. GCC has -cc1/-fc1 support too, but this patch does not address that.

### CODE REVIEW

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -2081,7 +2082,7 @@ bool Driver::HandleImmediateArgs(const Compilation &C) {
   if (C.getArgs().hasArg(options::OPT_dumpversion)) {
     // Since -dumpversion is only implemented for pedantic GCC compatibility, we
     // return an answer which matches our definition of __VERSION__.
-    llvm::outs() << CLANG_VERSION_STRING << "\n";
+    llvm::outs() << DriverVersion << "\n";
```
Comment (banach-space) [inline clang/lib/Driver/Driver.cpp:2085]: This change is not needed, is it?

**Review by banach-space — COMMENTED** [review]
Could this be implemented without any updates to Clang (beyond Options.td)?

**Comment by yuanfang-chen** [comment]
> Could this be implemented without any updates to Clang (beyond Options.td)?

The options could work by changing options.td only. However,  `CLANG_VERSION_STRING` and `FLANG_VERSION_STRING` could be different. This is to handle that.

**Comment by banach-space** [comment]
> > Could this be implemented without any updates to Clang (beyond Options.td)?
> 
> The options could work by changing options.td only. However, `CLANG_VERSION_STRING` and `FLANG_VERSION_STRING` could be different. This is to handle that.

You are adding a support for these new flags in `flang-new`, which is implemented in terms of `clangDriver` - this scenario would be very unlikely at the moment. And even if that was desired, you'd need to update Flang's driver too. I'd rather keep this to the required minimum. If you really want to keep this, you will need to update flang/tools/flang-driver/driver.cpp too.

**Comment by yuanfang-chen** [comment]
> You are adding a support for these new flags in `flang-new`, which is implemented in terms of `clangDriver` - this scenario (i.e. the two projects being out of sync) would be very unlikely at the moment (perhaps even impossible). And even if that was desired, you'd need to update Flang's driver too. I'd rather keep this to the required minimum. If you really want to keep this, you will need to update flang/tools/flang-driver/driver.cpp too.

Thanks for the explanation. I've updated the patch.

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

**Pull request merged**